File: src/main/java/com/jeffreyorazulike/simpletron/translator/SyntaxError.java
package com.jeffreyorazulike.simpletron.translator;

/**
 * SyntaxError
 */
public class SyntaxError extends RuntimeException {

    /**
     *
     */
    private static final long serialVersionUID = 7838378171908019726L;

    public SyntaxError() {
        this("Your code isn't well written");
    }

    public SyntaxError(String message) {
        super(message);
    }

    public SyntaxError(int line, String message) {
        this(String.format("Syntax Error on line %02d, %s.", line, message));
    }
}

File: translator/src/main/kotlin/dev/logickoder/simpletron/translator/syntax/SyntaxError.kt
package dev.logickoder.simpletron.translator.syntax

/**
 *
 * A message containing what is wrong with a line of code in simpletron basic language
 *
 * @property message what this error is all about
 */
class SyntaxError(message: String) : RuntimeException(message) {
    companion object {
        const val UNKNOWN_ERROR = "unknown error occurred"
    }
}